packages/providers/email-mailgun/src/__tests__/convert-provider-options.test.ts

```diff
@@ -50,7 +48,8 @@ describe('@strapi/provider-email-mailgun', () => {
     it('successfully creates a new Mailgun client', () => {
       const defaults = {
         username: 'api',
-      };
+      } as Options;
```

Comment:
Why are we typecasting here? 🤔 this could cause issues down the line in the test file when we're not sure about something...

Reply:
I'll see if I can change that without too much effort

Reply:
So considering the dynamic nature of how the options are built I don't think it's worth spending the time making this work perfectly as we will drop the old backward compat in v5 so the type will be straight forward. 

I moved the casting to the client creation instead to make this a tad better still